kroxylicious/src/main/java/io/kroxylicious/proxy/internal/filter/FetchResponseTransformationFilter.java

```diff
@@ -94,12 +100,13 @@ context.<MetadataResponseData> sendRequest(MetadataRequestData.HIGHEST_SUPPORTED
                         }
                         applyTransformation(context, fetchResponse);
                         LOGGER.debug("Forwarding original Fetch response");
-                        context.forwardResponse(header, fetchResponse);
+                        future.complete(new ResponseFilterResultImpl(header, fetchResponse, false));
```

Comment:
Just diving in not knowing how much of this is WIP :smile: Currently usages like this are causing kroxylicious to block up forever.

The sequence goes like:
1. we call `sendRequest` which writes some bytes to the channel (maybe this flows all the way to write to the socket, unsure?)
2. we return the `Future<ResponseFilterResult>` from `onFetchRespons`e.
3. FilterHandler blocks on the Future so the channel can't do any more work, the netty IO loop is blocked
4. We don't read the response from the broker or do more work on this channel (because the IO loop is blocked), so the `sendRequest` CompletionStage can never complete and we're deadlocked.

Reply:
Yeah, you are seeing a work in progress.